pygraylog/graylogapi.py

```diff
@@ -31,7 +36,13 @@ def method(**kwargs):
             return self._(name)
 
     def build_auth_header(self):
-        payload = self.username + ':' + self.password
+        
+        # use the api_key if it is been provided
+        if self.api_key is not None:
+            payload = self.api_key + ':token'
```

Comment:
👍  this looks like the way Graylog constructs the token header, http://docs.graylog.org/en/2.2/pages/configuration/rest_api.html